src/Avalonia.Controls/Flyouts/MenuFlyoutPresenter.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Avalonia.Controls.Generators;
+using Avalonia.Controls.Platform;
+using Avalonia.Controls.Primitives;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    public class MenuFlyoutPresenter : MenuBase
+    {
+        public MenuFlyoutPresenter()
+            :base(new DefaultMenuInteractionHandler(true))
+        {
+
+        }
+
+        public override void Close()
+        {
+            // DefaultMenuInteractionHandler calls this
+            var host = this.FindLogicalAncestorOfType<Popup>();
+            if (host != null)
+            {
+                host.IsOpen = false;
+            }
+        }
+
+        public override void Open()
+        {
```

Comment:
Probably you can save MenuFlyout instance inside of MenuFlyoutPresenter, when it's created.
And show this flyout from parent logical element, if it exists.
But I am not sure in what cases this method can be called @grokys .

ContextMenu tries to implement this method using saved list of attached controls https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ContextMenu.cs#L255

Reply:
Or you can use FindLogicalAncestorOfType<Popup> in same way, as it was used here in Close method

Reply:
`Open` is abstract from `MenuBase` so it had to be included, but from what I can tell it was never called from `DefaultMenuInteractionHandler`, so I think the only way would be if you called it directly. IMO, ignoring should be ok as one normally shouldn't interact with `MenuFlyoutPresenter`. Perhaps throwing a `NotSupportedException` or logging a warning might be better than just an ignore comment though.